### ISSUE TITLE
`manual_assert` false-positive for `else if`

### ISSUE THREAD

**Issue by DDtKey** [rust-lang/rust-clippy#9594]
### Summary

I believe `manual_assert`shouldn't check `else if` branches.
It could be long-chains of `if - else if- ... N ... - else` statements, so `assert` would duplicate statements & probably make it unreadable 

### Lint Name

manual_assert

### Reproducer

I tried this code:

```rust
#![deny(clippy::pedantic)]

fn main() {
    let cond_1 = false;
    let cond_2 = true;
    
    if cond_1 {
        println!("ok");
    } else if cond_2 {
        panic!("err")
    }
}

```

I saw this happen:

```
  --> src/main.rs:9:12
   |
9  |       } else if cond_2 {
   |  ____________^
10 | |         panic!("err")
11 | |     }
   | |_____^ help: try: `assert!(!cond_2, "err");`
```

I expected to see this happen:
if - else if  chains could be rather large, so I believe this lint should ignore `else if` branches 

### Version

```text
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-apple-darwin
release: 1.64.0
LLVM version: 14.0.6
```
